schemas/com.snowplowanalytics.snowplow.badrows/adapter_failures/jsonschema/1-0-0

```diff
@@ -0,0 +1,243 @@
+{
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Schema for a bad row resulting from adapter failures",
+  "self": {
+    "vendor": "com.snowplowanalytics.snowplow.badrows",
+    "name": "adapter_failures",
+    "format": "jsonschema",
+    "version": "1-0-0"
+  },
+  "type": "object",
+  "properties": {
+    "failure": {
+      "type": "object",
+      "description": "Information regarding the adapter violations",
+      "properties": {
+        "timestamp": {
+          "type": "string",
+          "description": "Timestamp at which the failure occurred",
+          "format": "date-time"
+        },
+        "vendor": {
+          "type": "string",
+          "description": "Vendor of the adapter that processed this payload",
+          "maxLength": 64
+        },
+        "version": {
+          "type": "string",
+          "description": "Version of the adapter that processed this payload",
+          "maxLength": 16
+        },
+        "messages": {
+          "type": "array",
+          "description": "List of failure messages associated with the adapter failures",
+          "items": {
+            "anyOf": [
+              {
+                "type": "object",
+                "description": "Error which was internal to the adapter regarding its input data",
+                "properties": {
+                  "field": {
+                    "type": "string",
+                    "description": "Field which did not meet the adapter's expectations",
+                    "maxLength": 64
+                  },
+                  "value": {
+                    "type": [ "string", "null" ],
+                    "description": "Stringified representation of the value which did not meet expectations"
+                  },
+                  "expectation": {
+                    "type": "string",
+                    "description": "Expectation which was not met",
+                    "maxLength": 256
```

Comment:
Not really. But my reasoning was that:

1. Nothing should really validate bad rows against their schemas
2. Most of our current warehouses don't have a notion of varchar size
3. One that have (Redshift) truncates these properties (and doesn't support JSON arrays, so not sure it even makes sense)
4. In the end, no matter what value we choose - there's always a chance it isn't enough
5. In absolute most of the cases this should be enough

TL;DR all these `maxLength`s are 99% informational

Reply:
> Nothing should really validate bad rows against their schemas
I had never thought about that!

Reply:
I agree, this is not obvious. But if you think about it:

1. Property-based tests in snowplow-badrows make us very confident that data conforms these schemas
2. If a bad row is in invalid in its consumer (e.g. was written not by snowplow-badrows) then it should be something "completely invalid", like missing required property (maybe multiple ones) rather than "slightly invalid" like exceeded string length
3. Both "completely invalid" and "slightly invalid" bad rows would end up in some kind of dead end and we likely are not interested in reasons and don't expect this to happen at all in a common case.

Reply:
Makes sense!